src/EFCore.PG/Extensions/NpgsqlNetworkExtensions.cs

```diff
@@ -0,0 +1,1180 @@
+﻿#region License
+
+// The PostgreSQL License
+//
+// Copyright (C) 2016 The Npgsql Development Team
+//
+// Permission to use, copy, modify, and distribute this software and its
+// documentation for any purpose, without fee, and without a written
+// agreement is hereby granted, provided that the above copyright notice
+// and this paragraph and the following two paragraphs appear in all copies.
+//
+// IN NO EVENT SHALL THE NPGSQL DEVELOPMENT TEAM BE LIABLE TO ANY PARTY
+// FOR DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES,
+// INCLUDING LOST PROFITS, ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS
+// DOCUMENTATION, EVEN IF THE NPGSQL DEVELOPMENT TEAM HAS BEEN ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//
+// THE NPGSQL DEVELOPMENT TEAM SPECIFICALLY DISCLAIMS ANY WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS
+// ON AN "AS IS" BASIS, AND THE NPGSQL DEVELOPMENT TEAM HAS NO OBLIGATIONS
+// TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+
+#endregion
+
+using System;
+using System.Net;
+using System.Net.NetworkInformation;
+using System.Runtime.CompilerServices;
+using JetBrains.Annotations;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.EntityFrameworkCore
+{
+    /// <summary>
+    /// Provides extension methods supporting operator translation for PostgreSQL network types.
+    /// </summary>
+    /// <remarks>
+    /// See: https://www.postgresql.org/docs/current/static/functions-net.html
+    /// </remarks>
+    public static class NpgsqlNetworkExtensions
+    {
+        #region RelationalOperators
+
+        /// <summary>
+        /// Determines whether an <see cref="IPAddress"/> is less than another <see cref="IPAddress"/>.
+        /// </summary>
+        /// <param name="_">The <see cref="DbFunctions"/> instance.</param>
+        /// <param name="inet">The left-hand inet.</param>
+        /// <param name="other">The right-hand inet.</param>
+        /// <returns>
+        /// True if the <see cref="IPAddress"/> is less than the other <see cref="IPAddress"/>; otherwise, false.
+        /// </returns>
+        /// <exception cref="NotSupportedException">
+        /// This method is only intended for use via SQL translation as part of an EF Core LINQ query.
+        /// </exception>
+        public static bool LessThan([CanBeNull] this DbFunctions _, IPAddress inet, IPAddress other)
+            => throw ClientEvaluationNotSupportedException();
```

Comment:
As discussed elsewhere, let's throw the standard `NotSupportedException` rather than this.

Reply:
This is actually just a [helper method](https://github.com/austindrenski/Npgsql.EntityFrameworkCore.PostgreSQL/blob/3ca6977533dfd7b8475e678ed1d681d8f3044992/src/EFCore.PG/Extensions/NpgsqlNetworkExtensions.cs#L1175-L1176) now:

```c#
static NotSupportedException ClientEvaluationNotSupportedException(
    [CallerMemberName] string method = default)
    => new NotSupportedException($"{method} is only intended for use via SQL translation as part of an EF Core LINQ query.");
```

Reply:
Ohh ok...